pkg/scheduler/cache/event_handlers.go

```diff
@@ -1261,3 +1264,126 @@ func (sc *SchedulerCache) setCSIResourceOnNode(csiNode *sv1.CSINode, node *v1.No
 		node.Status.Capacity[resourceName] = quantity
 	}
 }
+
+func (sc *SchedulerCache) getNodeNamesByRegexMatchSelector(selector *networktopov1alpha1.RegexMatch) []string {
+	// Get all node names from sc.Nodes that match the regex pattern
+	var matchingNodes []string
+
+	// Compile the regular expression
+	re, err := regexp.Compile(selector.Pattern)
+	if err != nil {
+		klog.Errorf("Failed to compile regex pattern: %v. Err = %v", selector.Pattern, err)
+	}
+
+	// Iterate over the nodes in the cache
+	for nodeName := range sc.Nodes {
+		if re.MatchString(nodeName) {
+			matchingNodes = append(matchingNodes, nodeName)
+		}
+	}
+
+	return matchingNodes
+}
+
+func (sc *SchedulerCache) getHyperNodeNamesByRegexMatchSelector(selector *networktopov1alpha1.RegexMatch) []string {
+	// Get all hypernode names from sc.HyperNodes that match the regex pattern
+	var matchingHyperNodes []string
+
+	// Compile the regular expression
+	re, err := regexp.Compile(selector.Pattern)
+	if err != nil {
+		klog.Errorf("Failed to compile regex pattern: %v. Err = %v", selector.Pattern, err)
```

Comment:
How about just return? Because the following iteration is meaningless if regex format was incorrect.